Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -661,24 +660,6 @@ WarpXParticleContainer::DepositCharge (amrex::Vector<std::unique_ptr<amrex::Mult
 #else
         ignore_unused(do_rz_volume_scaling);
 #endif
-
-        // Exchange guard cells
-        if (!local) rho[lev]->SumBoundary( m_gdb->Geom(lev).periodicity() );
-    }
-
-    // Now that the charge has been deposited at each level,
-    // we average down from fine to crse
```

Comment:
Note that an apparent advantage of the interpolation function below is that it does rely on the existence of the `rho_cp` `MultiFab`, and only touches the `rho_fp` `MultiFab` (which are passed as argument to this function under the name `rho`), while the function `SyncRho` does rely on the existence of `rho_cp`.

However, in practice, the function below *does* allocate a temporary `MultiFab` (`coarsened_fine_data`), which is essentially equivalent to `rho_cp`. Therefore, I don't think that we will see significant changes in performance (at least for a single level of refinement).

We could later optimize the `SyncRho` function to take into account the electrostatic case.